config/src/lib.rs

```diff
@@ -901,6 +903,95 @@ impl Config {
 
         Ok(())
     }
+
+    /// List the data in the foundry cache
+    pub fn list_foundry_cache() -> eyre::Result<Cache> {
+        if let Some(cache_dir) = Config::foundry_cache_dir() {
+            let mut cache = Cache { chains: vec![] };
+            if let Ok(entries) = cache_dir.as_path().read_dir() {
+                for entry in entries.flatten() {
```

Comment:
also filter by file type

Reply:
I filtered by type when i first wrote it but when refactoring i noticed i could remove the if before `if let Ok(entries) = cache_dir.as_path().read_dir()`, this line would do the work since we don't do anything if it's a file, so it kept the same functionality (as far as i can tell) while reducing code nesting.

If this is not idiomatic i'll be happy to change it, but i personally did not find it easier to read with the additional if

Reply:
you can simply `.flatten().filter(...)` by file type